app/code/core/Mage/Sales/Model/Order/Pdf/Abstract.php

```diff
@@ -366,15 +366,17 @@ protected function insertOrder(&$page, $obj, $putOrderId = true)
 
         if (!$order->getIsVirtual()) {
```

Comment:
```suggestion
        if (!$order->getIsVirtual() && isset($shippingAddress) && is_iterable($shippingAddress)) {
```

Reply:
that was my original idea but, since there was the `$this->y` and we're talking about PDF generation, I don't want to mess with anything just to avoid processing one line of code

Reply:
If there is no shipping address `$this->y` isnt changed inside the loop (?)

(not tested)